python/pyspark/sql/dataframe.py

```diff
@@ -3082,6 +3082,13 @@ def unpivot(
         When no "id" columns are given, the unpivoted DataFrame consists of only the
         "variable" and "value" columns.
 
+        The `values` columns must not be empty so at least one value must be given to be unpivoted.
+        When `values` is `None`, all non-id columns will be unpivoted.
+
+        Note: A column that is referenced by an id column expression is considered an id column
+        itself. For instance `col("id") * 2` references column `id`, so `id` is considered an
+        id column and not a value column.
```

Comment:
@HyukjinKwon is this behavior the same as pandas?

Reply:
No, it is not, `pyspark.pandas.DataFrame.melt` compares value column candidates literally with id column names.

Reply:
Is this to match other SQL databases?

Reply:
No, see https://github.com/apache/spark/pull/37407/files#r972844049.